background.js

```diff
@@ -1,74 +1,62 @@
 const SETTINGS_KEY = "HATABS_SETTINGS";
-let last_on = null;
 
-function checkTabs() {
-  const settings = JSON.parse(localStorage.getItem(SETTINGS_KEY) || "{}");
-
-  const { host, apikey, device, sites } = settings;
-
-  if (!host || !apikey || !device || !sites) {
-    console.log("invalid settings for HA-Tabs");
-  }
-
-  const siteary = (sites || "")
-    .split(";")
-    .map((x) => x.trim())
-    .filter((x) => x !== "");
+// Define the api variable for compatibility
+const api = typeof browser === "undefined" ? chrome : browser;
 
-  browser.tabs.query({}).then((tabs) => {
-    let on = false;
-
-    const activeTab = tabs.find((tab) => tab.active);
-
-    if (activeTab) {
-      on = siteary.some((site) => {
-        return activeTab.url.includes(site);
-      });
+function checkTabs() {
+  // Use chrome.storage API to access the settings
+  chrome.storage.local.get([SETTINGS_KEY], (result) => {
```

Comment:
i think you need to use api.xx instead of chrome.xxx since you've defined api in line 4?

Reply:
you are probably using chrome, but i don't think this will work on Firefox

Reply:
Ohh you're right, I'm using it in a chrome based browser. Didn't think of firefox 🥲